xgo.go

```diff
@@ -233,12 +233,24 @@ func pullDockerImage(image string) error {
 func compile(image string, config *ConfigFlags, flags *BuildFlags, folder string) error {
 	// If a local build was requested, find the import path and mount all GOPATH sources
 	locals, mounts, paths := []string{}, []string{}, []string{}
+	var usesModules bool
 	if strings.HasPrefix(config.Repository, string(filepath.Separator)) || strings.HasPrefix(config.Repository, ".") {
 		// Resolve the repository import path from the file path
 		config.Repository = resolveImportPath(config.Repository)
 
+		// Check if the repo uses go 1.11 modules by determening if it has a go.mod file
+		if _, err := os.Stat(config.Repository + "/go.mod"); !os.IsNotExist(err) {
```

Comment:
FYI, a go.mod file can live in any directory, not just the repository root.  (But this code will probably work fine for most repos.)  Ideally the `go/build` or `golang.org/x/tools/go` package would provide this information, but they don't seem to have been updated for modules yet.  You should recursively check parent directories for go.mod, starting at the package to be built.

Reply:
Oh, I didn't knew that. Thanks for pointing it out, I'll update the patch accordingly.

Reply:
Regarding the treatment of `go.mod` in places other than the repository's root in my commits in this PR:

I don't feel this use case warranted supporting in this PR. As the go modules docs even state (https://github.com/golang/go/wiki/Modules#gomod), most often `go.mod` will be in the repository root. As it stands, I'm happy for others to add support for arbitrary `go.mod` paths, but I recommend that it be part of a new issue/feature request with it's own compact PR.